File: src/Server/Infrastructure/Persistence/Entities/Attachment.cs
using Server.Core.Common;
using Server.Infrastructure.Identity;

namespace Server.Infrastructure.Persistence.Entities;

public class Attachment : BaseEntity
{
    public string FileName { get; set; } = string.Empty;

    public string OriginalFileName { get; set; } = string.Empty;

    public string FileExtension { get; set; } = string.Empty;

    public string MimeType { get; set; } = string.Empty;

    public long FileSize { get; set; }

    public string StoragePath { get; set; } = string.Empty;

    public string RelatedEntityType { get; set; } = string.Empty;

    public Guid RelatedEntityId { get; set; }

    public Guid? UploadedBy { get; set; }

    public string? Description { get; set; }

    public string? Tags { get; set; }

    public ApplicationUser? Uploader { get; set; }
}

File: src/client/src/types/attachment.ts
export interface AttachmentDetail {
  id: string;
  fileName: string;
  originalFileName: string;
  fileExtension: string;
  mimeType: string;
  fileSize: number;
  description?: string | null;
  tags?: string | null;
  uploadedByName: string;
  createdAt: string;
  downloadUrl: string;
}

export interface AttachmentListItem {
  id: string;
  originalFileName: string;
  fileExtension: string;
  fileSize: number;
  uploadedByName: string;
  createdAt: string;
}

export interface UploadAttachmentRequest {
  file: File;
  relatedEntityType: string;
  relatedEntityId: string;
  description?: string | null;
  tags?: string | null;
}

export interface UpdateAttachmentRequest {
  description?: string | null;
  tags?: string | null;
}
